tests/versioning_test.go

```diff
@@ -245,37 +243,51 @@ func (s *versioningIntegSuite) TestMaxTaskQueuesPerBuildIdEnforced() {
 
 func (s *versioningIntegSuite) testWithMatchingBehavior(subtest func()) {
```

Comment:
I think this structure is a little bit unintuitive:

```go
var overrides map[dynamicconfig.Key]any
wrapper := func() {
  // we override config here as s.T() is unique per subtest.
  // if we override values in the outer scope then these will be bound to the outer
  // lifetime and cause the bug fixed by https://github.com/temporalio/temporal/pull/5046
  for k, v := range overrides {
    dc.OverrideValue(s.T(), k, v)
  }

  subtest()
}

for _, forceForward := range []bool{false, true} {
  for _, forceAsync := range []bool{false, true} {
    // assign to overrides map
    s.Run(name, wrapper)
  }
}
```
What about something like this instead:

```go
for _, forceForward := range []bool{false, true} {
  for _, forceAsync := range []bool{false, true} {
    s.Run(name, func() {
      // set overrides directly (they will have the right s.T() now)
      subtest()
    })
  }
}
```

Reply:
Fixed. It's still unwieldy because we use these values to generate the test name, but it's at least all handled in the same scope